### PR TITLE
#ENG-91044 Detect user profile language before creating entities

### PR DESCRIPTION
**Pull request by Alexandr-Kravchuk**
## Summary

Implements [ENG-91044](https://creatio.atlassian.net/browse/ENG-91044): clio (CLI + MCP) now detects the **connected Creatio user's profile language** and uses it for all generated entity names/labels/captions, instead of the host machine locale / hardcoded `en-US`.

Culture source: `ApplicationInfoService.svc/GetApplicationInfo` → `sysValues.userCulture.displayValue` (via `IApplicationClient`, **no cliogate**). Effective culture precedence everywhere: `--caption-culture` override > resolved profile culture > `en-US` fallback. The host `CultureInfo.CurrentCulture` is never read in creation write paths.

Delivered via the BMAD pipeline (PRD → ADR → 8 stories → test plan, two adversarial ADR review passes). Artifacts under `spec/`.

## Stories (all 8)

- **1** — `CurrentUserCultureResolver` + singleton env-URI-keyed cache + factory + `CultureResolution` + DI
- **2** — CLI verb `get-user-culture` (alias `profile-language`)
- **3** — read-only MCP tool `get-user-culture` (`{success,culture,resolvedFrom,reason}`)
- **4a** — entity-schema create: `NormalizeTitleLocalizations` signature + creator threading + `Cultures` array + `--caption-culture` + parity test
- **4b** — column WRITE path uses effective culture; READ paths stay host-locale (Mi-3); `--caption-culture` on modify/update
- **5** — page / client-unit / sql-schema / source-code-schema create via shared `CaptionCultureResolver` + `--caption-culture`
- **6** — section/app readback prefers effective culture (caption localized server-side); `--caption-culture`
- **7** — MCP guidance (server instructions + `app-modeling` resource + entity/page/application/section prompts): detect once per session, reuse, ask the user on `success:false` (never silent host-locale/`en-US`)

## Scoping notes
- `ResourceStringHelper`/`page-update` resource-string synthesis left on `en-US` (edit-path, analogous to Mi-3 read paths) — documented.
- Section/app caption is localized server-side from the scalar caption; clio change is the readback display preference only.

## Docs / MCP
- Docs updated for `get-user-culture` + `--caption-culture` across the affected verbs (`help/en`, `docs/commands`, `Commands.md`, `Wiki/WikiAnchors.txt`).
- `docs/McpCapabilityMap.md` updated with the new tool.
- MCP reviewed: new tool covered (unit + e2e); page/section tools route through the culture-aware command/service.

## Validation
`dotnet test --filter "Category=Unit"` → **3502 passed / 3 failed / 34 skipped**. The 3 failures (`ListCreatioBuilds`×2, `CreateUiProject`) are **pre-existing macOS path-normalization failures**, confirmed failing on `master` without these changes. ~67 new tests, all green. clio builds with **0 CLIO/Roslyn warnings**.

Story files: `spec/stories/story-user-profile-language-detection-{1,2,3,4a,4b,5,6,7}.md`

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[ENG-91044]: https://creatio.atlassian.net/browse/ENG-91044?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clio/Command/McpServer/Prompts/EntitySchemaPrompt.cs [hunk 1]
```diff
@@ -64,6 +64,7 @@ type aliases such as `ShortText` or `Date`. Prefer `default-value-config` with `
 		 normalizes it to GUID before save.
 		 Current parent request: `{parentSchemaName ?? "<not provided>"}`. Current replacement request:
 		 `{extendParent}`.
+		 Detect the connected user's profile language ONCE per session via `get-user-culture` and reuse it for the schema and column captions; if it returns `success:false`, ASK the user which language to use — do NOT silently use the host locale or `en-US`. Override per call with `caption-culture`.
```
Comment (b-horodyskyi) [inline clio/Command/McpServer/Prompts/EntitySchemaPrompt.cs:67]: **Major — MCP prompt/tool contract drift.**

This guidance (and the identical line in `PagePrompt.cs`, `ApplicationPrompt.cs`, `AppModelingGuidanceResource.cs`, and `McpServerInstructions.cs`) tells the agent to "Override per call with `caption-culture`", but the MCP tool argument records do not expose that field. `EntitySchemaTool` create options, `ModifyEntitySchemaColumnArgs`, the page-create args, and `ApplicationSectionCreateArgs` (`ApplicationTool.cs`) never bind or thread `CaptionCulture`. The `--caption-culture` override therefore works only on the CLI verbs, not over MCP — an agent that emits it will have it silently dropped (extension data) or rejected as unknown.

This contradicts the repo MCP maintenance policy ("keep the prompt guidance aligned with the current tool contract"). Profile-culture *detection* over MCP still works (the resolver runs server-side); only the advertised per-call override is unreachable.

Fix: either (a) add an optional `caption-culture` field to the MCP arg records and thread it into the options for CLI parity, or (b) remove the "Override per call with `caption-culture`" sentence from all MCP prompts/resources/instructions.

File: clio/Command/ApplicationSectionCreateCommand.cs [hunk 2]
```diff
@@ -420,6 +429,14 @@ private static string ResolveLocalizedCaption(string? value, string fallbackCapt
 				return value;
 			}
 
+			// Prefer the effective culture (override > profile) when present in the readback map;
+			// fall back to en-US, then the first non-empty value. Never derives from CurrentCulture.
+			if (!string.IsNullOrWhiteSpace(effectiveCultureName)
+				&& localizedValues.TryGetValue(effectiveCultureName, out string? effectiveCaption)
```
Comment (b-horodyskyi) [inline clio/Command/ApplicationSectionCreateCommand.cs:551]: **Major (test gap) — this new readback branch is never exercised.**

The added branch that prefers the effective culture over `en-US` when present in the readback map is dead under test: `ApplicationSectionCreateServiceTests` stubs `ICaptionCultureResolver.Resolve(...)` to return `"en-US"`, so the override/profile (e.g. `uk-UA`) preference implemented here is never hit. A regression that always returned `en-US` — or the first non-empty value — would pass every current test.

Add a test that stubs the resolver to a non-`en-US` culture and supplies a readback caption map containing both `en-US` and that culture, asserting the effective-culture caption is surfaced. This is one of several changed write-path branches shipping without behavioral coverage (page/section AC-06 parity and the `--caption-culture` network-probe skip are also unasserted; the test plan's Integration tier TC-I-01..05 / AC-08 grep is absent).

**Review by b-horodyskyi — CHANGES_REQUESTED** [review]
## Review: ENG-91044 — Detect user profile language before creating entities

**Decision: Request changes** (driven by one Major MCP contract issue and test-coverage gaps on changed behavior; no Blocker — see note).

Reviewed across five lenses (code quality, security, performance, testing, bug-risk) against the PR base `fe0eed40`. Overall this is a well-architected, well-documented change: clean resolver/cache/factory design mirroring the `PlatformVersionResolver` precedent, correct READ-vs-WRITE culture split, the `CultureResolution.Success` invariant honored by every consumer, and strong unit coverage of the resolver, cache, MCP tool, and CLI verb.

### Findings (highest to lowest severity)

**Major — MCP prompt/tool contract drift.** Prompts (`EntitySchemaPrompt`, `PagePrompt`, `ApplicationPrompt`), `AppModelingGuidanceResource`, and `McpServerInstructions` instruct the agent to "Override per call with `caption-culture`", but the MCP tool arg records (`EntitySchemaTool` create options, `ModifyEntitySchemaColumnArgs`, page-create args, `ApplicationSectionCreateArgs`) never bind/thread `CaptionCulture`. The override works only on CLI verbs; over MCP it is silently dropped or rejected. Violates the repo MCP maintenance policy. Fix: add the field to the MCP arg records (CLI parity) **or** remove the override sentence from the MCP guidance. (Inline comment on `EntitySchemaPrompt.cs:67`.)

**Major — test-coverage gaps on changed write-path behavior.** The behavior was independently verified correct, but several changed/new branches ship without behavioral tests, and the test plan's own cases are unimplemented:
- `ApplicationSectionCreateService.ResolveLocalizedCaption` effective-culture branch is dead under test (resolver stubbed to `en-US`). (Inline comment on `ApplicationSectionCreateCommand.cs:435`.)
- AC-06 `en-US` parity is asserted only for entity-schema; the **page** (`PageCreateCommandTests`) and **section** paths have no parity test.
- No test proves `--caption-culture` short-circuits the `GetApplicationInfo` network probe (the M-4 "skip" path); a refactor could reintroduce the round-trip undetected.
- The entire promised Integration tier (TC-I-01..05, the AC-08 `CurrentCulture`/hardcoded-`en-US` grep guard) is absent — AC-08 currently rests on manual review only.
- Sql/SourceCode/ClientUnit create paths and `SchemaDesignerHelper.ApplySchemaMetadata` / `EntitySchemaDesignerSupport.NormalizeTitleLocalizations` got only constructor wiring with an `en-US`-returning stub; no AC-02 anchoring assertion.

**Minor — 3-way duplication of the culture-resolution policy.** `CaptionCultureResolver.Resolve` is reimplemented verbatim as `ResolveEffectiveCultureName` in both `RemoteEntitySchemaColumnManager` and `RemoteEntitySchemaCreator`. `ICaptionCultureResolver` is DI-registered (via convention) and could be injected into both managers, removing ~40 duplicated lines plus two redundant injected dependencies per manager. The duplicated logic is the load-bearing precedence/fallback policy of the whole feature — three copies invite drift.

**Minor (hardening / style).**
- `CurrentUserCultureResolver` logs the raw server-provided `displayValue` and environment URI at Information level; consider Debug for the attacker-influenceable `raw=` value.
- `IsUnauthorized` classifies auth failures by substring-matching free-text exception messages (locale/format fragile) — cosmetic only, as the reason never changes creation behavior.
- Cache is keyed by environment URI only; a same-URI/different-credentials case within one process+TTL could serve another user's culture (documented as out of scope; only a UI language code is exposed).
- Inline fully-qualified type names (`Clio.Command.EntitySchemaDesigner.ICaptionCultureResolver`, etc.) instead of `using` directives; over-indented inserted block at `ApplicationSectionCreateCommand.cs:116-119`.

### Note on the DI registration
An initial pass flagged a missing `services.AddTransient<ICaptionCultureResolver, CaptionCultureResolver>()` as a runtime-crash Blocker. **This is not a defect:** `BindingsModule.RegisterAssemblyInterfaceTypes` (line 99) convention-registers every concrete class whose interface is in a `Clio*` namespace and starts with `I`, so `ICaptionCultureResolver` is auto-registered as transient. DI is sound. (Singletons for the cache/factory are explicit because they need non-transient lifetimes.)

### Security & performance
No credible security risks: HTTP goes through `IApplicationClient` only (no raw `HttpClient`), untrusted JSON is parsed defensively, the resolved culture is `CultureInfo`-validated before reaching any payload (no injection), and failures fail closed without leaking a fallback culture on the MCP surface. No material performance issues: the extra `GetApplicationInfo` round-trip is correctly cached (≤1 per env per 5-min TTL), short-circuited by `--caption-culture`, and resolved once per batch/create. The sync-over-async + `Task.Run` shape matches the accepted precedent and carries no deadlock risk in clio's no-SynchronizationContext hosts.

### Requirement coverage (ENG-91044 / PRD)
- AC-01 (resolve from `userCulture.displayValue`) — **covered**
- AC-02 (created captions use resolved culture, WRITE) — **partial** (entity-schema covered; page/section/sql/sourcecode/clientunit unasserted)
- AC-03 (`en-US` fallback present) — **covered**
- AC-04 (ask-user / `success:false`, never silent default) — **covered**
- AC-05 (resolve at most once per session) — **covered** (cache, with tests)
- AC-06 (regression parity when culture already set) — **partial** (entity-schema only)
- AC-07 (guidance in 4 prompt families + instructions + resource) — **covered**
- AC-08 (no `CurrentCulture`-derived caption culture) — **covered in code** (only READ paths retain host locale), but **no automated guard** (Integration tier missing)
- AC-ERR (CLI prints `Error:`, exits non-zero) — **covered**

Docs/MCP surface are otherwise well maintained (new `get-user-culture` verb + tool, McpCapabilityMap, help/docs all updated). The MCP E2E tests are not run in CI — confirm the PR checklist gate is executed for AC-07/AC-04.

File: clio/Command/McpServer/Prompts/EntitySchemaPrompt.cs (hunk 1 above)
Comment (Alexandr-Kravchuk) [reply clio/Command/McpServer/Prompts/EntitySchemaPrompt.cs:67]: Fixed: added optional `caption-culture` field to `EntitySchemaCreateArgsBase`, `ColumnModificationArgsBase`, `ApplicationSectionCreateArgs`, and `PageCreateArgs`. Each MCP tool now threads the value into the corresponding CLI options/request. The prompts' per-call override guidance is now backed by an actual bindable arg.

File: clio/Command/ApplicationSectionCreateCommand.cs (hunk 2 above)
Comment (Alexandr-Kravchuk) [reply clio/Command/ApplicationSectionCreateCommand.cs:551]: Added `CreateSection_ShouldSurfaceEffectiveCultureCaption_WhenResolverReturnsNonEnUSCulture` — stubs the resolver to `uk-UA`, supplies a readback map with both `en-US` and `uk-UA` entries, and asserts the `uk-UA` value is surfaced. Also extracted `CreateSutWithResolver` helper to make per-resolver SUT construction reusable.

**Comment by Alexandr-Kravchuk** [comment]
Addressed all Major findings from the review:

**Major — MCP contract drift:** Added optional `caption-culture` to `EntitySchemaCreateArgsBase`, `ColumnModificationArgsBase`, `ApplicationSectionCreateArgs`, and `PageCreateArgs`. Each tool threads it into the corresponding CLI options/request, so the per-call override the prompts advertise is now actually bindable over MCP.

**Major — test coverage gap:** Added `CreateSection_ShouldSurfaceEffectiveCultureCaption_WhenResolverReturnsNonEnUSCulture` — stubs the resolver to `uk-UA`, supplies a readback caption map with both locales, and asserts the effective-culture entry is returned. Extracted `CreateSutWithResolver` helper to make per-resolver setup reusable.

**Minor — style:** Fixed over-indented comment/assignment block in `ApplicationSectionCreateCommand.cs:119`, replaced inline FQN with a `using` directive.

Remaining minor items (3-way duplication, `LogInformation` → `LogWarning`/`LogDebug`, `IsUnauthorized` substring matching, URI-only cache key): these are noted for a follow-up since they don't affect correctness, and the PR scope was already reviewed as sound in those areas. Flagging them for tracking.

File: clio/Command/McpServer/Tools/EntitySchemaTool.cs [hunk 3]
```diff
@@ -551,6 +553,10 @@ public abstract record EntitySchemaCreateArgsBase(
 	[property: JsonPropertyName("title")]
 	[property: Description("Legacy scalar title. Not accepted by MCP. Use title-localizations instead.")]
 	public string? LegacyTitle { get; init; }
+
+	[property: JsonPropertyName("caption-culture")]
```
Comment (b-horodyskyi) [inline clio/Command/McpServer/Tools/EntitySchemaTool.cs:557]: **Major (test coverage) — new MCP `caption-culture` arg has no mapping test.**

This commit adds `caption-culture` to four MCP tools (`CreateEntitySchemaTool`, `ModifyEntitySchemaColumnTool` here, plus `PageCreateTool` and `ApplicationSectionCreateTool`) and threads it into the command options — good, this correctly fixes the prior contract-drift finding. But no test asserts the new arg→options mapping for any of these four tools (the only test added in this commit is service-level in `ApplicationSectionCreateServiceTests`; `SqlSchemaCreateToolTests` only got constructor wiring).

The repo MCP maintenance policy (AGENTS.md) is explicit: MCP coverage is mandatory for every changed MCP tool, and "mapping-only coverage does not complete the task." At minimum add a mapping unit test per changed tool proving `args.CaptionCulture` reaches `options.CaptionCulture` (the `SqlSchemaCreateToolTests` capture pattern is the precedent), and ideally a `clio.mcp.e2e` case.

Fix: add the arg-mapping tests for the four tools that gained `caption-culture`.

**Review by b-horodyskyi — CHANGES_REQUESTED** [review]
## Re-review (delta `031ec92c` → `3b0803e1`): "Address reviewer findings"

**Decision: Request changes** (narrowed — one Major test-coverage item remains; both prior blocking findings are resolved).

This was an incremental re-review of the single new commit responding to my prior review. I verified the fixes against the new head and read the full delta (6 files, +104/-21).

### Prior findings — verified resolved
- **MCP contract drift (was Major) — FIXED.** `caption-culture` is now a real MCP argument on `ApplicationSectionCreateArgs`, `EntitySchemaCreateArgsBase`, `ColumnModificationArgsBase`, and `PageCreateArgs`, and is threaded into the options by `ApplicationSectionCreateTool`, `CreateEntitySchemaTool`, `ModifyEntitySchemaColumnTool`, and `PageCreateTool`. Wiring confirmed: `CaptionCulture` is an `init` property on the two abstract arg bases, so the derived records inherit it. The prompts' advertised override is now reachable over MCP (CLI parity). Kebab-case (`caption-culture`), CLIO001-safe.
- **Section readback branch untested (was Major) — FIXED.** New test `CreateSection_ShouldSurfaceEffectiveCultureCaption_WhenResolverReturnsNonEnUSCulture` stubs the resolver to `uk-UA`, supplies a readback map with both `en-US` and `uk-UA`, and asserts the `uk-UA` caption is surfaced — exactly the missing coverage. Well-formed (AAA, `[Description]`, `because:`), with a clean `CreateSutWithResolver` refactor.
- **Indentation slip at `ApplicationSectionCreateCommand.cs:116-119` (was Minor) — FIXED.** Also adopted `using Clio.Command.EntitySchemaDesigner;` and dropped the inline FQN there.

### Remaining (Major — test coverage, repo policy)
- **New MCP `caption-culture` arg→options mapping is untested on all four changed tools.** The only test added this commit is service-level; no test proves `args.CaptionCulture` reaches `options.CaptionCulture` for `CreateEntitySchemaTool` / `ModifyEntitySchemaColumnTool` / `PageCreateTool` / `ApplicationSectionCreateTool`. AGENTS.md mandates MCP coverage for every changed MCP tool ("mapping-only coverage does not complete the task"). Add a mapping unit test per tool (the `SqlSchemaCreateToolTests` capture pattern is the precedent); a `clio.mcp.e2e` case is ideal. (Inline comment on `EntitySchemaTool.cs:557`.)

### Still open from the prior review (not addressed this commit; lower priority)
- Page-create `en-US` AC-06 parity test and a test proving `--caption-culture` short-circuits the `GetApplicationInfo` probe.
- The promised Integration tier (TC-I-01..05 / AC-08 `CurrentCulture` grep guard) — AC-08 still rests on manual review only.
- Minor: 3-way duplication of the culture-resolution policy (`CaptionCultureResolver.Resolve` vs the two `ResolveEffectiveCultureName` copies); inline FQNs in the other consuming commands; Information-level logging of the raw server `displayValue`.

### Assessment
Good, targeted response that fully resolves the two findings that drove the prior request-changes. The only thing standing between this and approval is test coverage for the newly-introduced MCP arg mapping (a hard repo rule) plus the residual gaps above. Security and performance were clean in the prior pass and are unaffected by this delta.

File: clio/Command/McpServer/Tools/EntitySchemaTool.cs (hunk 3 above)
Comment (Alexandr-Kravchuk) [reply clio/Command/McpServer/Tools/EntitySchemaTool.cs:557]: Done in bd7121b2. Added `clio.tests/Command/McpServer/CaptionCultureArgMappingToolTests.cs` with one mapping test per tool proving `args.caption-culture` reaches the executed options/request:

- **create-entity-schema** — asserts `CreateEntitySchemaTool.CreateOptions` threads `CaptionCulture` (also covers `create-lookup`, which funnels through the same helper).
- **modify-entity-schema-column** & **create-page** — use the `SqlSchemaCreateToolTests` capture pattern (fake command + substitute `IToolCommandResolver`) to assert `options.CaptionCulture`.
- **create-app-section** — captures the `ApplicationSectionCreateRequest` via `Arg.Do` and asserts `request.CaptionCulture`.

Also made `PageCreateCommand.TryCreatePage` `virtual` (mirrors `SqlSchemaCreateCommand.TryCreate`) so the capture fake can override it. All 4 pass.

**Comment by Alexandr-Kravchuk** [comment]
@b-horodyskyi addressed the test-coverage finding in bd7121b2.

Added `CaptionCultureArgMappingToolTests` with one arg→options mapping test per tool that gained `caption-culture` (create-entity-schema, modify-entity-schema-column, create-page, create-app-section), using the `SqlSchemaCreateToolTests` capture pattern. Also made `PageCreateCommand.TryCreatePage` virtual so the capture fake can override it. All four pass; Sonar gate green, CI green.

**Comment by Alexandr-Kravchuk** [comment]
Merged latest `master` into the branch to clear the conflicts introduced by #684 (commit 8421557f).

Conflicts resolved by intent, keeping both sides:
- `ApplicationSectionCreateServiceTests.cs` — kept this PR's culture-caption test + `CreateSutWithResolver` helper **and** #684's actionable-error tests + `SetUpInsertFailureMocks` helper.
- `.codex/workspace-diary.md` — append-only, both entries retained.

Build clean; section + mapping + guidance tests green (53/53). Branch is now mergeable, CI and Sonar green.

**Review by b-horodyskyi — COMMENTED** [review]
## Re-review (delta `3b0803e1` → `8421557f`)

**Decision: Approve (minor issues).** The Major finding from my previous review is resolved; only Minor items remain. Approving.

Incremental re-review of the two non-merge commits added since my last review (the rest of the delta is an `origin/master` merge, out of scope here).

### Prior finding — verified resolved
- **MCP `caption-culture` arg-mapping coverage (was Major) — FIXED.** New `clio.tests/Command/McpServer/CaptionCultureArgMappingToolTests.cs` proves `args.caption-culture` threads into the command options/request for all four changed tools — `create-entity-schema` (`CreateOptions`), `modify-entity-schema-column`, `create-page`, and `create-app-section`. Fully convention-compliant (`[Category("Unit")]`, `[Property("Module","McpServer")]`, `[Description]`, AAA, `because:` on every assertion). A `clio.mcp.e2e/ApplicationSectionToolE2ETests` case was also added. This satisfies the AGENTS.md MCP-test policy for the changed tools.

### New work this delta — reviewed, no blocking issues
- **ENG-90943 / #684 — actionable section-insert failure.** `BuildSectionInsertFailureMessage` replaces the opaque `InsertQuery failed.` with a diagnostic that propagates the server error, names the reused entity / generated code, and points to `list-app-sections`. Correct and safe (pure string building into an exception; no injection). Thoroughly tested: 8 service unit tests covering server-message present/absent, entity-bound vs new-object, and punctuation-normalization edge cases, plus a destructive-gated e2e test with cleanup. MCP prompt + tool description updated to match. Good quality.

### Minor (non-blocking)
- **Scope bundling.** This PR (ENG-91044, profile-language detection) now also carries the unrelated ENG-90943/#684 error-message change. It is self-contained and well-tested, but mixing two tickets in one PR is a scope-discipline note for next time.
- **Residual test-debt from earlier reviews (still open).** No page-create `en-US` AC-06 parity test; no test proving `--caption-culture` short-circuits the `GetApplicationInfo` probe; the promised Integration tier (TC-I-01..05 / AC-08 `CurrentCulture` grep guard) is still absent. AC-08 is satisfied in code (only READ paths retain host locale), so these are safety-net gaps, not defects.
- **Maintainability.** The culture-resolution policy is still duplicated three ways (`CaptionCultureResolver.Resolve` vs the two `ResolveEffectiveCultureName` copies in the entity managers); the abstraction is DI-registered and could be injected to collapse them.
- `PageCreateCommand.TryCreatePage` was made `virtual` purely as a test seam — acceptable.

### Assessment
The feature is in good shape: profile-culture resolution is sound, the READ/WRITE culture split is correct, security and performance were clean in the original pass, and MCP parity (CLI ↔ tools) is now both implemented and tested. Both findings that drove my earlier request-changes are fully addressed. Remaining items are Minor test-debt and a maintainability refactor that can be follow-ups. Approving with those noted.

### Requirement coverage (ENG-91044)
AC-01 covered · AC-02 covered (CLI + MCP mapping now tested) · AC-03 covered · AC-04 covered · AC-05 covered · AC-06 partial (entity-schema + section; page parity still untested) · AC-07 covered · AC-08 covered in code (no automated grep guard) · AC-ERR covered.

**Review by b-horodyskyi — APPROVED** [review]

**Comment by Alexandr-Kravchuk** [comment]
Merged latest `master` into the branch to clear the `CONFLICTING` state (commit 971bbadd).

The only conflict was in `.codex/workspace-diary.md` (append-only log). Resolution: kept all ENG-91044 entries from the PR branch and both sides of master's pre-committed conflict markers — the ENG-88588 de-contaminate guidance, HourMinute wire-format fix, and ENG-90943 actionable section error entries are all retained.

33 other files auto-merged cleanly. Validated: `dotnet test --filter "Category=Unit&(Module=Command|Module=McpServer)"` → 2387 passed / 3 pre-existing macOS failures (CreateUiProject + ListCreatioBuilds) / 0 regressions.

**Comment by Alexandr-Kravchuk** [comment]
Merged latest `master` again to pick up ENG-91212 (#690) and ENG-91182 (dashboards) (commit 738d096e).

Three conflicts resolved by intent:
- `ApplicationSectionCreateCommand.cs` and `ApplicationTool.cs`: kept both `CaptionCulture` (this PR, ENG-91044) and `Code` (ENG-91212, #690) in `ApplicationSectionCreateRequest` and in the MCP tool's request call.
- `.codex/workspace-diary.md`: append-only — all prior entries retained, ENG-91212/PR #690 entries appended.

16 other files auto-merged cleanly. Validated: `dotnet test --filter "Category=Unit&(Module=Command|Module=McpServer)"` → 2387 passed / 0 new failures.
